examples/common/src/framework.rs

```diff
@@ -100,7 +102,8 @@ async fn setup<E: Example>(title: &str) -> Setup {
         use winit::platform::windows::WindowBuilderExtWindows;
         builder = builder.with_no_redirection_bitmap(true);
     }
-    let window = builder.build(&event_loop).unwrap();
+    static WINDOW: OnceLock<Window> = OnceLock::new();
+    let window = WINDOW.get_or_init(|| builder.build(&event_loop).unwrap());
```

Comment:
I know this is a bit sketchy, but as we are using `winit::event_loop::EventLoop::spawn()` for Wasm, we actually require a `'static` `Window`.

Reply:
This basically reverts #4588 now, which isn't needed now because we never drop `Window` at this point.

So if we want to *improve* this:
- Use some self-referential shenanigans. E.g. [`self_cell`](https://crates.io/crates/self_cell) or [`oroboros`](https://crates.io/crates/ouroboros).
- Separate `Window` creation from `setup()` so can hold it beyond `run()`, solving the lifetime issue because `Surface` is moved into `run()`, which makes it drop before `Window`.

We don't have to do anything though, this is just an improvement for users trying to learn how to do this from that code.

Reply:
Resolved in https://github.com/gfx-rs/wgpu/pull/4597/commits/3d57ebadabb1e356eeebed07d0d5634cdffb3cec.